apm-agent-core/src/main/java/co/elastic/apm/bci/bytebuddy/SoftlyReferencingTypePoolCache.java

```diff
@@ -71,14 +72,13 @@ public void run() {
             return TypePool.CacheProvider.Simple.withObjectType();
         }
         lastAccess.set(System.currentTimeMillis());
-        classLoader = classLoader == null ? ClassLoader.getSystemClassLoader() : classLoader;
+        classLoader = classLoader == null ? getBootstrapMarkerLoader() : classLoader;
         SoftReference<TypePool.CacheProvider> cacheProviderRef = cacheProviders.get(classLoader);
-        while (cacheProviderRef == null || cacheProviderRef.get() == null) {
+        if (cacheProviderRef == null || cacheProviderRef.get() == null) {
```

Comment:
I think that if `cacheProviderRef.get() == null`, you should turn off caching for a while altogether...
If JVM is struggling with memory you don't want to make it even harder with all these SoftReferences hadnling

Reply:
Could also mean the JVM has had memory issues and now everything is fine again. It's also unclear for how long to turn off caching. For the first version of this, I'd like to avoid adding additional complexity.